modules/wordads/wordads.php

```diff
@@ -481,14 +507,26 @@ function get_ad( $spot, $type = 'iponweb' ) {
 	 *
 	 * @since 5.7
 	 */
-	function get_ad_snippet( $section_id, $height, $width, $location = '', $css = '' ) {
+	public function get_ad_snippet( $section_id, $height, $width, $location = '', $css = '' ) {
 		$this->ads[] = array(
 			'location' => $location,
 			'width'    => $width,
 			'height'   => $height,
 		);
-		$ad_number   = count( $this->ads ) . '-' . uniqid();
 
+		if ( self::is_amp() ) {
+			$amp_section_id = self::get_amp_section_id();
+			$site_id = $this->params->blog_id;
+			return <<<HTML
+			<amp-ad width="$width" height="$height"
+			    type="pubmine"
+			    data-siteid="$site_id"
+			    data-section="$amp_section_id">
+			</amp-ad>
+HTML;
+		}
+
+		$ad_number = count( $this->ads ) . '-' . uniqid();
```

Comment:
Using `uniqid` breaks the AMP plugin's post-processor cache. Better to use https://developer.wordpress.org/reference/functions/wp_unique_id/ or a polyfill.

Reply:
Though I suppose this won't run in AMP anyway since it's after the `return`?

Reply:
> Though I suppose this won't run in AMP anyway since it's after the return?
That's correct.